tests/awardFinancialMixed.csv

```diff
@@ -2,11 +2,11 @@ allocationtransferagencyidentifier,agencyidentifier,beginningperiodofavailabilit
 0,19,2016,2016,,113,0,ABCD,ABCD,123456798,12-34-56-78-90-AS-DF-AB-XZ-YW,Overseas Programs,2,1255,D,6500,6500,26000,6500,6500,6500,6500,6500,26000,6500,6500,6500,6500,13000,52000,6500,26000,6500,6500,6500,6500,6500,26000,6500,6500,6500,6500,6500,6500,6500,6500,6500,26000
 ,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
 0,19,2016,2016,,113,0,ABCD,ABCD,123456800,12-34-56-78-90-AS-DF-AB-XZ-YW,Overseas Programs,ABCD,ABCD,D,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD,ABCD
-1900,7200,10000,10000,XFA,10950,9999,ABCD,ABCD,123456801,12-34-56-78-90-AS-DF-AB-XZ-YW,Overseas Programs,23456,25412,D,6500,6500,26000,6500,6500,6500,6500,6500,26000,6500,6500,6500,6500,13000,52000,6500,26000,6500,6500,6500,6500,6500,26000,6500,6500,6500,6500,6500,6500,6500,6500,6500,26000
+1900,7200,10000,10000,XFA,10950,9999,ABCD,ABCD,123456801,12-34-56-78-90-AS-DF-AB-XZ-YW,Overseas Programs,23456,25412,X,6500,6500,26000,6500,6500,6500,6500,6500,26000,6500,6500,6500,6500,13000,52000,6500,26000,6500,6500,6500,6500,6500,26000,6500,6500,6500,6500,6500,6500,6500,6500,6500,26000
 0,19,2016,2016,,113,1,ABCD,ABCD,123456802,12-34-56-78-90-AS-DF-AB-XZ-YW,Overseas Programs,2,1255,,6501,6502,6503,6504,6505,6506,6507,6508,6509,6510,6511,6512,6513,6514,6515,6516,6517,6518,6519,6520,6521,6522,6523,6524,6525,6526,6527,6528,6529,6530,6531,6532,6533
 0,19,2016,2016,,113,0,ABCD,ABCD,123456803,12-34-56-78-90-AS-DF-AB-XZ-YW,Overseas Programs,2,1255,D,6500,6500,26000,6500,6500,6500,6500,6500,26000,6500,6500,6500,6500,13000,52000,6500,26000,6500,6500,6500,6500,6500,26000,6500,6500,6500,6500,,6500,6500,6500,6500,26000
-0,19,2013,2016,B,999,0,ABCD,ABCD,123456804,12-34-56-78-90-AS-DF-AB-XZ-YW,Overseas Programs,2,1255,R,,,,,,,,,,,,,,,,,,,,,,,,,,,,6500,,,,,
-0,19,2016,2016,,113,0,ABCD,ABCD,123456805,12-34-56-78-90-AS-DF-AB-XZ-YW,Overseas Programs,2,1255,D,6500,6500,26000,6500,6500,6500,6500,6500,26000,6500,6500,6500,6500,13000,52000,6500,26000,6500,6500,6500,6500,6500,26000,6500,6500,6500,6500,6500,6500,6500,6500,6500,26000
+0,19,2013,2016,B,999,0,ABCD,ABCD,123456804,12-34-56-78-90-AS-DF-AB-XZ-YW,Overseas Programs,2,1255,RR,,,,,,,,,,,,,,,,,,,,,,,,,,,,6500,,,,,
+0,19,2016,2016,,113,0,ABCD,ABCD,123456805,12-34-56-78-90-AS-DF-AB-XZ-YW,Overseas Programs,2,2255,R,6500,6500,26000,6500,6500,6500,6500,6500,26000,6500,6500,6500,6500,13000,52000,6500,26000,6500,6500,6500,6500,6500,26000,6500,6500,6500,6500,6500,6500,6500,6500,6500,26000
```

Comment:
The program activity file has 'Q' and 'RR', and has '1255' paired with 'R', along with just '255' paired with both 'D' and 'R'.  All of those are records that have the ussgl numbers present.  Does that cover the cases you're looking for here, or do you have a different test in mind?